### ISSUE TITLE
Satellite crashes running under Windows 11

### ISSUE THREAD

**Issue by phillipivan** [bitfocus/companion-satellite#141]
![image](https://github.com/user-attachments/assets/e103cf1b-cd1f-43ff-8fce-f71bed124fac)

We have encountered crashes when running Satellite Version 1.8.1 under Windows 11, with a Stream deck XL connected. See attached error message.

Please let me know what other information I can provide to assist. We will check the event log for further clues tomorrow.

**Comment by Julusian** [comment]
Does this happen when connecting or disconnecting the streamdeck, or randomly?
How consistently/often is this happening?

**Comment by phillipivan** [comment]
We have only seen it occur once to date, without obvious cause, so for the time being 'randomly'. Happy to do some investigative work if you can point in a worthwhile direction.

**Comment by phillipivan** [comment]
We have had this occur again overnight, here is the message from the event log:

> Application popup: Companion Satellite.exe - Application Error : The instruction at 0x00007FFFA8C1B48A referenced memory at 0x0000000000000048. The memory could not be read.

**Comment by phillipivan** [comment]
Updated we corrected version number (actually 1.7.5, thought that machine was on 1.8.1)

**Comment by Julusian** [comment]
Does windows event view say anything more on this issue? perhaps giving an indicate of a dll or executable filename that triggered it?

**Comment by phillipivan** [comment]
My previous comment had the only relevant entry we could find in windows event viewer last time it occurred, which doesn't really add anything beyond the error message dialog box.

**Comment by JuddiP** [comment]
Hi @Julusian,

Copying in further application event details from Windows Event Viewer (from 04-09-2024) -

**4/09/2024 9:40:29 AM**

- Faulting application name: Companion Satellite.exe, version: 1.7.5.0, time stamp: 0x656fb32b
- Faulting module name: 550c65d3-0e4e-4a70-9e1f-00129bcc3de2.tmp.node, version: 0.0.0.0, time stamp: 0x651b0c23
- Exception code: 0xc0000005
- Fault offset: 0x000000000004b48a
- Faulting process id: 0x0x2C64
- Faulting application start time: 0x0x1DAF5263C924422
- Faulting application path: C:\Program Files\Companion Satellite\Companion Satellite.exe
- Faulting module path: \\?\C:\Users\hanlocal\AppData\Local\Temp\550c65d3-0e4e-4a70-9e1f-00129bcc3de2.tmp.node
- Report Id: 67f480f6-601c-42f1-9929-f2559f9051ca
- Faulting package full name: 
- Faulting package-relative application ID: 

It is interesting to note that we currently have another application that runs on these PCs that allows us to control and preview our ingest servers that is experiencing a memory leak. Do you think that could possibly be related? These events are typically occurring a little while after reported resource exhaustion events -

![Image](https://github.com/user-attachments/assets/2ee0ae3c-a338-4110-8c35-6b94f13d1e36)

The above was reported last night, and subsequently Satellite was closed out (with no application error event, oddly enough).

**Comment by JuddiP** [comment]
Hey mate,

Unfortunately it looks like that temp file no longer exists on the machine.

Is there anything else I can do to assist?

Check the faulting module path file again as soon as it occurs?

**Comment by Julusian** [comment]
Yeah when it next happens, the same information as provided above and a copy of that `.tmp.node` file would be good.

I'm not sure what else to do. Maybe I could make a build of satellite which does more logging to try and narrow down where it is happening? Ill have a think about that

**Comment by Julusian** [comment]
I've added a new library which should produce better logs/errors when encountering a segfault. This is in 1.9.3.

**Comment by Julusian** [comment]
Is anyone still experiencing this?
Maybe the library I added back in october produces some further logging that could be helpful in figuring this out?

**Comment by phillipivan** [comment]
I don't think we have seen another crash in some time. Probably since we last reported one here.

We have stopped running the software @JuddiP referred to as having a memory leak. So perhaps this is a contributibg factor.
